### PR TITLE
21280  - Add restart reason code to psychometrician report

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**

### CODE REVIEW

File: admin/services/data-access/pupil-restart.data.service.js [hunk 1]
```diff
@@ -159,4 +159,24 @@ pupilRestartDataService.sqlMarkRestartAsDeleted = async (pupilId, userId) => {
...
+pupilRestartDataService.sqlFindByPupilIds = async (pupilIds) => {
+  if (!(Array.isArray(pupilIds) && pupilIds.length > 0)) {
+    throw new Error('No ids provided')
+  }
+  const select = `
+  SELECT *
+  FROM ${sqlService.adminSchema}.[pupilRestart]
+  `
+  const {params, paramIdentifiers} = sqlService.buildParameterList(pupilIds, TYPES.Int)
+  const whereClause = 'WHERE pupil_id IN (' + paramIdentifiers.join(', ') + ')'
```
Comment (GuyHarwood) [inline admin/services/data-access/pupil-restart.data.service.js]: inline would read nicer

File: admin/services/data-access/pupil-restart.data.service.js [hunk 2]
```diff
@@ -159,4 +159,24 @@ pupilRestartDataService.sqlMarkRestartAsDeleted = async (pupilId, userId) => {
...
+  const orderClause = 'ORDER BY createdAt ASC'
```
Comment (GuyHarwood) [inline admin/services/data-access/pupil-restart.data.service.js]: maybe add this to the inline?

File: admin/services/data-access/pupil-restart.data.service.js (hunk 1 above)
Comment (GeorgeChatzigiannis) [reply admin/services/data-access/pupil-restart.data.service.js]: @GuyHarwood should be better now

File: admin/services/data-access/pupil-restart.data.service.js [hunk 3]
```diff
@@ -159,4 +159,26 @@ pupilRestartDataService.sqlMarkRestartAsDeleted = async (pupilId, userId) => {
   return sqlService.modify(sql, params)
 }
 
+/**
+ * Find all active restarts based on pupil ids
+ * @param pupilIds
+ * @return {Promise<*>}
+ */
+pupilRestartDataService.sqlFindByPupilIds = async (pupilIds) => {
```
Comment (jon-shipley) [inline admin/services/data-access/pupil-restart.data.service.js]: This appears to be a test-developer function looking at the security (or rather the lack of the school parameter) - so do we want to separate it out into the psychometric report data service?

I started this pattern after discussion on the security fix bug.

Comment (GeorgeChatzigiannis) [reply admin/services/data-access/pupil-restart.data.service.js]: @jon-shipley done

**Comment by GeorgeChatzigiannis** [comment]
This is blocked until changes occur to link check with pupil restart

**Pull request closed without merging**